content/en/docs/concepts/services-networking/network-policies.md

```diff
@@ -312,6 +323,60 @@ The value of the label is the namespace name.
 While NetworkPolicy cannot target a namespace by its name with some object field, you can use the
 standardized label to target a specific namespace.
 
+## Pod lifecycle
+
+A NetworkPolicy is applied to a selected pod, when isolation rules are enforced
+for that pod. At that point pod can never establish an outbound connection or receive
+an inbound connection that is not allowed by the NetworkPolicy. Allow rules will be applied
+eventually after the isolation rules (that includes applying both at the same time).
+That means, that once the NetworkPolicy is applied to a pod, the pod should be able to handle situations,
+when it can't establish a connection that is allowed by an existing NetworkPolicy for some time.
```

Comment:
hm... this is all correct, but it's not very clear.

My attempt:

Because the network plugin may implement NetworkPolicy in a distributed manner (for example, with each node enforcing the policies only for pods on that node), it is possible that pods may see a slightly inconsistent view of network policies when the pod is first created, or when pods or policies change. For example, a newly-created pod that is supposed to be able to reach both Pod A on Node 1 and Pod B on Node 2 may find that it can reach Pod A immediately, but cannot reach Pod B until a few seconds later.

Network plugins should attempt to ensure that all policy rules are correctly configured as soon as possible after pod creation, but pods must be resilient against being started up with less network connectivity than expected (or in the worst case, no network connectivity at all, since if the network plugin can't guarantee that the rules are exactly right at pod startup, it must err on the side of making them too restrictive, to ensure that the pod can never establish an outbound connection or receive an inbound connection that is not allowed by NetworkPolicy).